sdk/src/azure/core/az_mqtt5_rpc_server_hfsm.c

```diff
@@ -337,7 +337,13 @@ static az_result waiting(az_event_policy* me, az_event event)
         }
 
         // parse the request details and send it to the application for execution
-        _az_RETURN_IF_FAILED(_handle_request(this_policy, recv_data));
+        if (az_result_failed(_handle_request(this_policy, recv_data)))
```

Comment:
There should never be a seg-fault: while I do agree with the fix to log in this case and ignore the message, inbound posts that cause faults should be handled by the MQTT implementation.

Reply:
Addressed in az_event_policy - before, if a message had failed to be sent inbound, it would send a failure message to the current policy, while the failure message should also be sent inbound (which will be a policy collection, unlike the current policy)